src/components/Scatterplot/Layout.js

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import AxisPlacement from '../AxisPlacement';
+import { axisPlacementType } from '../../utils/proptypes';
+
+const propTypes = {
+  chart: PropTypes.node.isRequired,
+  xAxis: PropTypes.node.isRequired,
+  yAxis: PropTypes.node.isRequired,
+  yAxisPlacement: axisPlacementType,
+};
+
+const defaultProps = {
+  yAxisPlacement: AxisPlacement.RIGHT,
+  contextChart: null,
+};
+
+const axisContainer = area => (axis, placement) => (
+  <div
+    key={area}
+    className="y-axis-container"
+    style={{
+      gridArea: area,
+      display: 'flex',
+      flexDirection: 'column',
+    }}
+  >
+    {React.cloneElement(axis, { yAxisPlacement: placement })}
+  </div>
+);
+
+const Layout = ({ chart, xAxis, yAxis, yAxisPlacement }) => {
+  const yAxes = [];
+  let areas;
+
+  const quote = s => `'${s}'`;
+
+  switch (yAxisPlacement) {
+    case AxisPlacement.LEFT:
+      areas = [
+        // formatting for readability
+        'yaxis chart',
+        '. xaxis',
+        '. context',
+      ]
+        .map(quote)
+        .join(' ');
+      yAxes.push(axisContainer('yaxis')(yAxis, yAxisPlacement));
+      break;
+    case AxisPlacement.BOTH: {
+      areas = [
+        // formatting for readability
+        'yaxis-left chart yaxis-right',
+        '. xaxis .',
+        '. context .',
+      ]
+        .map(quote)
+        .join(' ');
+      yAxes.push(axisContainer('yaxis-left')(yAxis, AxisPlacement.LEFT));
+      yAxes.push(axisContainer('yaxis-right')(yAxis, AxisPlacement.RIGHT));
+      break;
+    }
+    case AxisPlacement.RIGHT:
+    case AxisPlacement.UNSPECIFIED:
+    default:
+      areas = [
+        // formatting for readability
+        'chart yaxis',
+        'xaxis .',
+        'context .',
+      ]
+        .map(quote)
+        .join(' ');
+      yAxes.push(axisContainer('yaxis')(yAxis, yAxisPlacement));
+      break;
+  }
+  return (
```

Comment:
I could be wrong, but it seems that LineChart and Scatterplot layouts could be unified

Reply:
Yeah, but I'm okay with leaving them separate for now. If they remain the same, we can consolidate, but they might diverge over time.